src/include/optimizer/rule_query_rewrite.h

```diff
@@ -0,0 +1,73 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// rule_query_rewrite.h
+//
+// Identification: src/include/optimizer/rule_query_rewrite.h
+//
+// Copyright (c) 2015-18, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "optimizer/rule.h"
+
+#include <memory>
+
+namespace peloton {
+namespace optimizer {
+
+/**
+ * Implements transitive predicates generation for
+ * LogicalGet nodes:
+ *   a = b AND b = c -> a = b AND b = c AND a = c
+ */
+class TransitivePredicatesLogicalGet : public Rule {
+ public:
+  TransitivePredicatesLogicalGet();
+
+  bool Check(std::shared_ptr<OperatorExpression> plan,
```

Comment:
as you are declaring the same methods here and below for all of these classes, i'd recommend creating an abstract class that inherits from Rule so that you don't need to duplicate the declaration of the same `Check` and `Transform` methods